src/ComptimeInterpreter.zig

```diff
@@ -0,0 +1,1454 @@
+//! Hacky comptime interpreter, courtesy of midnight code run fuelled by spite;
+//! hope that one day this can use async... <33
+
+// TODO: DODify
+
+const std = @import("std");
+const ast = @import("ast.zig");
+const zig = std.zig;
+const Ast = zig.Ast;
+const analysis = @import("analysis.zig");
+const DocumentStore = @import("DocumentStore.zig");
+const ComptimeInterpreter = @This();
+
+const log = std.log.scoped(.comptime_interpreter);
+
+// TODO: Investigate arena
+
+allocator: std.mem.Allocator,
+document_store: *DocumentStore,
+handle: *const DocumentStore.Handle,
```

Comment:
If i remember correctly because of my changes in #702, Pointers to `DocumentStore.Handle` may get invalidated if you update the `DocumentStore` i would suggest storing the Handle URI with `DocumentStore.Uri` (alias for `[]const u8`)